doc/api/cli.md

```diff
@@ -121,6 +121,13 @@ added: v6.0.0
 
 Print stack traces for process warnings (including deprecations).
 
+### `--redirect-warnings=file`
```

Comment:
The idea for this came up during the collaborator summit and was specifically around finding a more intelligent way to handle deprecation warnings that are emitted when using tools like npm. When doing installs, it still may be necessary to capture `stderr` output as normal, but allow warnings to be deferred, captured, and processed in other ways.

This can be configured by environment variable. The envvar is included in this PR.

The file is created if it does not already exist.